server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -479,7 +480,14 @@ && isConflictException(executionResult.getFailure().getCause())
         }
         final BulkItemResponse response;
         if (isUpdate) {
-            response = processUpdateResponse((UpdateRequest) docWriteRequest, context.getConcreteIndex(), executionResult, updateResult);
+            final MappingLookup mappingLookup = context.getPrimary().mapperService().mappingLookup();
```

Comment:
As this is the `TransportShardBulkAction`, I don't think there's the possibility of having a null `MapperService` or `MappingLookup`. Do you think we should be defensive about that, or can we be sure that they're not null?

Reply:
Agreed, `MapperService` and `MappingLookup` should always be non-null here. Given we're getting this from an `IndexShard` instance, I don't think we need to be _that_ defensive; how about an `assert` so we catch it in tests if it's a problem?

Reply:
> how about an assert so we catch it in tests if it's a problem?

I like that, thanks Mike!